src/include/migraphx/shape.hpp

```diff
@@ -59,6 +59,43 @@ struct shape
     {
     };
 
+    struct dynamic_dimension
+    {
+        std::size_t min = 0;
+        std::size_t max = 0;
+        std::size_t opt = 0;
+
+        dynamic_dimension() = default;
+
+        dynamic_dimension(std::size_t i_min, std::size_t i_max, std::size_t i_opt)
+            : min(i_min), max(i_max), opt(i_opt)
```

Comment:
I made the constructor because emplace_back() doesn't work with the aggregate constructor until C++20. I'll refactor and avoid the emplace_back() defect.